airflow/providers/mongo/hooks/mongo.py

```diff
@@ -75,6 +74,10 @@ def __init__(self, mongo_conn_id: str = default_conn_name, *args, **kwargs) -> N
         self.client: MongoClient | None = None
         self.uri = self._create_uri()
 
+        self.allow_insecure = True
```

Comment:
I think allow_insecure should be  a separate extra, to alllow the user to come back to old behaviour. BTW. This is a breaking change so it neds to be explained in CHANGELOG for the provider.

Reply:
Do you mean to just add a new allow_insecure in extra and keep ssl as-is? Would that add any value?

Reply:
I believe `allow_insecure` only applies when `ssl` is used (plain http is always insecure). But in fact there were 2 states here:

* http (ssl = False)
* htps + allow_insecure (ssl = True)

What we need is 3 states

* http (ssl = False)
* htps + allow_insecure (ssl = True, allow_insecure=True)
* https  - (ssl = True, allow_insecure = False) this should be default when ssl

So we need separate `allow_insecure` flag with default = False